packages/block-library/src/navigation/edit.js

```diff
@@ -44,6 +47,38 @@ const LAYOUT = {
 	alignments: [],
 };
 
+function getBlockDOMNode( clientId, doc ) {
+	return doc.getElementById( 'block-' + clientId );
+}
+
+function getComputedStyle( node ) {
+	return node.ownerDocument.defaultView.getComputedStyle( node );
+}
+
+function detectColors( clientId, setColor, setBackground ) {
+	const colorsDetectionElement = getBlockDOMNode( clientId, document );
```

Comment:
Reading the code, this stands out a bit, do we use this approach anywhere else? Did we run into any trouble with using child refs?

Reply:
This is pretty much copy/pasted from `packages/block-library/src/button/color-edit.js`. I proposed some ideas around making that more extensible but the consensus was that it was better to do a customized implementation within the Navigation block.

I did find [this conversation](https://github.com/WordPress/gutenberg/commit/0c6e36977f32b5ea184cdb7e7e94bac3efb73b34#r43840820) about the original code.

I'll try to see if I can get the colors with a ref.

Reply:
Seems to be working with a ref!